File: Cliente/cliente-recaudos/src/app/shared/models/pagination.ts
import { DataReporte, IDataReporte } from "./dataReporte";
import { IRecaudo } from "./recaudo";

export interface IPagination {
    pageIndex: number;
    pageSize: number;
    count: number;
    data: IRecaudo[];
}

export interface IPaginationReporte {
  pageIndex: number;
  pageSize: number;
  count: number;
  dataObject: IDataReporte;
}
  
export class Pagination implements IPagination{
  pageIndex: number;
  pageSize: number;
  count: number;
  data: IRecaudo[] = [];
}

export class PaginationReporte implements IPaginationReporte{
  pageIndex: number;
  pageSize: number;
  count: number;
  dataObject: DataReporte;
}

File: Conteo y recaudo/Helpers/Pagination.cs
﻿using System.Collections.Generic;

namespace Conteo_y_recaudo.Helpers
{
    public class Pagination<T> where T : class
    {
        public Pagination(int pageIndex, int pageSize, int count, IReadOnlyList<T> data)
        {
            PageIndex = pageIndex;
            PageSize = pageSize;
            Count = count;
            Data = data;
        }

        public Pagination(int pageIndex, int pageSize, int count, T data)
        {
            PageIndex = pageIndex;
            PageSize = pageSize;
            Count = count;
            DataObject = data;
        }

        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public int Count { get; set; }
        public IReadOnlyList<T> Data { get; set; }
        public T DataObject { get; set; }
    }
}
